odoo/addons/base/models/ir_cron.py

```diff
@@ -98,17 +98,22 @@ def _process_jobs(cls, db_name):
                 if not jobs:
                     return
                 cls._check_modules_state(cron_cr, jobs)
-                job_ids = tuple([job['id'] for job in jobs])
 
-                while True:
-                    job = cls._acquire_one_job(cron_cr, job_ids)
+                for job in jobs:  # acquire one by one
+                    job_id = job['id']
+                    try:
+                        job = cls._acquire_one_job(cron_cr, (job_id,))
+                    except psycopg2.errors.SerializationFailure:
+                        _logger.warning("job %s has been processed by other thread", job_id)
+                        continue
```

Comment:
That does not look right to me.  You cannot simply continue, since the transaction corresponding to `cron_cr` is in failure mode from this point on.

IMHO that failure should be caught outside of the loop, and either retry the whole stuff right away, or give up from this point on and retry later.

Reply:
I've understanded. Retry to aquired cronjob rather than continue error may be the right way. 
Is `check` function in `odoo.service.model` to decorate `_acquire_one_job` or `_process_jobs` the right way? Or code retry again？